src/cmd-cloud-prune

```diff
@@ -0,0 +1,272 @@
+        ref_date = datetime.datetime.now() - relativedelta(months=int(duration_in_months))
+
+        print(f"Pruning resources of type {action} older than {duration_in_months} months ({ref_date.date()}) on stream {stream}")
+        # Enumerating in reverse to go from the oldest build to the newest one
+        for index, build in enumerate(reversed(builds_json_data["builds"])):
+            build_id = build["id"]
+            if "policy-cleanup" in build.keys():
+                # If we have already pruned the specified resources for this
+                # build as per builds.json, we skip through it.
+                if action in build["policy-cleanup"]:
+                    print(f"Build {build_id} has already had {action} pruning completed")
+                    continue
+            # Assuming only FCOS versioning as of now
+            timestamp = parse_fcos_version(build_id)[1]
+            build_date = datetime.datetime(int(timestamp[0:4]), int(timestamp[4:6]), int(timestamp[-2:]))
+
+            if build_date >= ref_date:
+                break
+            for arch in build["arches"]:
+                meta_url = f"/builds/{build_id}/{arch}/meta.json"
+                meta_json = get_json_from_s3(s3_client, bucket, prefix + meta_url)
+                if meta_json is None:
+                    print(f"Failed to get meta.json for {build_id} for {arch}")
+                    return
+                # Check if the build has any cloud platform that are not GC supported yet
+                images = get_supported_images(meta_json, unsupported, supported)
+                currentBuild = Build(
+                    id=build_id,
+                    timestamp=timestamp,
+                    images=images,
+                    arch=arch,
+                )
+                match action:
+                    case "cloud-uploads":
+                        # Prunes only AWS and GCP at the moment
+                        delete_cloud_resources(currentBuild, cloud_config, args.dry_run)
+                        if not args.dry_run:
+                            build.setdefault("policy-cleanup", []).append(action)
+                            builds_json_data["builds"][index] = build
+                            builds_json_data["timestamp"] = datetime.datetime.now(pytz.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+                    # Not implemented yet
+                    case "build":
+                        return NotImplementedError
+                        # print(f"Deleting key {prefix}{build.id} from bucket {bucket}")
+                        # Delete the build's directory in S3
+                        # S3().delete_object(args.bucket, f"{args.prefix}{str(currentBuild.id)}")
+
+    if not args.dry_run:
+        with open("builds/builds.json", "w") as json_file:
+            json.dump(builds_json_data, json_file, indent=2)
+        # This copies the local builds.json and updates the S3 bucket version.
+        if args.upload_builds_json:
```

Comment:
Not quite. This needs to be a different mode entirely. So basically:
- `cosa cloud-prune --policy ...` prunes resources, updates `builds.json` locally.
- `cosa cloud-prune --upload-builds-json` does not do any pruning. It downloads the latest `builds.json`, merges the modifications from the local `builds.json` into the latest, and uploads to S3.